rancher2/data_source_rancher2_project.go

```diff
@@ -0,0 +1,90 @@
+package rancher2
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/rancher/norman/types"
+)
+
+func dataSourceRancher2Project() *schema.Resource {
+	return &schema.Resource{
+		Exists: dataSourceRancher2ProjectExists,
+		Read:   dataSourceRancher2ProjectRead,
+
+		Schema: map[string]*schema.Schema{
+			"uuid": {
+				Description: "UUID of the project",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"cluster_id": {
+				Description: "ID of the cluster to whom the project belongs",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"name": {
+				Description: "Name of the project",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"description": {
+				Description: "Description of the project",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+		},
+	}
+}
+
+func dataSourceRancher2ProjectExists(d *schema.ResourceData, meta interface{}) (bool, error) {
+	client, err := meta.(*Config).ManagementClient()
+	if err != nil {
+		return false, err
+	}
+
+	clusterID := d.Get("cluster_id").(string)
+	name := d.Get("name").(string)
+
+	projects, err := client.Project.List(&types.ListOpts{
+		Filters: map[string]interface{}{
+			"cluster_id": clusterID,
+			"name":       name,
+		},
+	})
+	if err != nil {
+		return false, err
+	}
+
+	cnt := len(projects.Data)
+	if cnt <= 0 {
+		return false, nil
+	}
+	if cnt > 1 {
+		return false, fmt.Errorf("more than one project with specified name (\"%s\") found: %d", name, cnt)
+	}
+
+	// Only one project returned? Great...
+	return true, nil
+}
+
+func dataSourceRancher2ProjectRead(d *schema.ResourceData, meta interface{}) error {
```

Comment:
This function seems wrong, it's calling settings functions. Could you please review it??

Reply:
Oh dear. I'll revisit my PR.
I did push the wrong changeset... my bad.